lms/djangoapps/lms_xblock/test/test_runtime.py

```diff
@@ -125,18 +125,13 @@ def setUp(self):
         self.course_id = CourseLocator("org", "course", "run")
         self.user = UserFactory.create()
 
-        def mock_get_real_user(_anon_id):
-            """Just returns the test user"""
-            return self.user
-
         self.runtime = LmsModuleSystem(
             static_url='/static',
             track_function=Mock(),
             get_module=Mock(),
             render_template=Mock(),
             replace_urls=str,
-            course_id=self.course_id,
```

Comment:
Oops, yes it is.. But unfortunately, the tests still passed because [`UserCourseTag` model doesn't require `course_id` to be set](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/models.py#L156) .

Have [required both `user` and `course_id` to be set before initializing the `UserTagService`](https://github.com/edx/edx-platform/pull/28440/files#diff-2f075571d3c19527241117ab2c4688b58a1ae60799a4d136acbb466a1ccdadd0R157-R161), and re-added this line.

Reply:
Hmm, I am assuming both should always be present since the LMS runtime only works in the context of a course for a specific user?